configure.ac

```diff
@@ -5077,6 +5079,62 @@ then
     AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_DES_ECB -DWOLFSSL_SIGNER_DER_CERT"
 fi
 
+# curl Support
+AC_ARG_ENABLE([curl],
+    [AS_HELP_STRING([--enable-curl],[Enable curl (default: disabled)])],
+    [ ENABLED_CURL=$enableval ],
+    [ ENABLED_CURL=no ]
+    )
+# curl support requires all the features enabled within this conditional.
+if test "$ENABLED_CURL" = "yes"
+then
+    if test "x$ENABLED_DES3" = "xno"
+    then
+        ENABLED_DES3="yes"
+    fi
+
+    if test "x$ENABLED_ALPN" = "xno"
+    then
+        ENABLED_ALPN="yes"
+    fi
+
+    if test "x$ENABLED_WOLFSSH" = "xno"
+    then
+        ENABLED_WOLFSSH="yes"
+    fi
+
+    if test "x$ENABLED_OPENSSLEXTRA" = "xno" && test "x$ENABLED_OPENSSLCOEXIST" = "xno"
+    then
+        ENABLED_OPENSSLEXTRA="yes"
+    fi
+
+    if test "x$ENABLED_CRL" = "xno"
+    then
+        ENABLED_CRL="yes"
+        AM_CFLAGS="$AM_CFLAGS -DHAVE_CRL"
+    fi
+
+    if test "x$ENABLED_OCSP" = "xno"
+    then
+        ENABLED_OCSP="yes"
+        AM_CFLAGS="$AM_CFLAGS -DHAVE_OCSP"
+    fi
+
+    if test "x$ENABLED_CERTIFICATE_STATUS_REQUEST" = "xno"
+    then
+        ENABLED_CERTIFICATE_STATUS_REQUEST="yes"
+        AM_CFLAGS="$AM_CFLAGS -DHAVE_TLS_EXTENSIONS -DHAVE_CERTIFICATE_STATUS_REQUEST"
+    fi
+
+    if test "x$ENABLED_CERTIFICATE_STATUS_REQUEST_V2" = "xno"
+    then
+        ENABLED_CERTIFICATE_STATUS_REQUEST_V2="yes"
+        AM_CFLAGS="$AM_CFLAGS -DHAVE_TLS_EXTENSIONS -DHAVE_CERTIFICATE_STATUS_REQUEST_V2"
+    fi
+
+    AM_CFLAGS="$AM_CFLAGS -DWOLFSSL_DES_ECB"
```

Comment:
@bagder , I think the need for this flag and ENABLED_DES3 did not exist a few months ago.  Has something changed recently to require these?

Reply:
Note that without these, curl's configure step finishes without any errors and then `make all` fails with linker errors.

Reply:
I don't know what specific change that caused this. Maybe this? https://github.com/curl/curl/commit/e12dc2dd977c1e5f8d05e681d8d31f4fc124f6f9

I figure my use of `--enable-all` in the CI build hid when it happened.

Reply:
Hi @bagder tracing it back I think the 'SSL_CTX_set1_curves_list' issue was something that happened around 4 months and 8 days ago. Am updating our tests temporarily to have --enable-all also, then switching over to this --enable-curl build once the PR is complete.

Reply:
Amusing, because [d10a0af2f](https://github.com/curl/curl/commit/d10a0af2f) which brought the use of that function was done by @anhu  :grin: ...